menas/src/main/scala/za/co/absa/enceladus/rest/services/RunService.scala

```diff
@@ -15,14 +15,125 @@
 
 package za.co.absa.enceladus.rest.services
 
-import org.springframework.beans.factory.annotation.Autowired
+import java.util.UUID
+
+import org.joda.time.format.DateTimeFormat
+import org.springframework.beans.factory.annotation.{Autowired, Value}
 import org.springframework.stereotype.Service
+import za.co.absa.atum.model.{Checkpoint, ControlMeasure, RunStatus}
+import za.co.absa.enceladus.model.{Run, SplineReference}
+import za.co.absa.enceladus.rest.exceptions.{NotFoundException, ValidationException}
+import za.co.absa.enceladus.rest.models.Validation
 import za.co.absa.enceladus.rest.repositories.RunMongoRepository
 
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
 @Service
 class RunService @Autowired()(runMongoRepository: RunMongoRepository)
   extends ModelService(runMongoRepository) {
 
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  @Value("${za.co.absa.enceladus.spline.urlTemplate}")
+  val splineUrlTemplate: String = null
+
+  def getAllLatest(): Future[Seq[Run]] = {
+    runMongoRepository.getAllLatest()
+  }
+
+  def getByStartDate(startDate: String): Future[Seq[Run]] = {
+    Try(DateTimeFormat.forPattern("dd-MM-yyyy").parseLocalDate(startDate)) match {
+      case _: Success[_] => runMongoRepository.getByStartDate(startDate)
+      case _: Failure[_] =>
+        val validation = Validation().withError("startDate", s"must have format dd-MM-yyyy: $startDate")
+        throw ValidationException(validation)
+    }
+  }
+
+  def getRun(datasetName: String, datasetVersion: Int, runId: Int): Future[Run] = {
+    runMongoRepository.getRun(datasetName, datasetVersion, runId).map {
+      case Some(run) => run
+      case None      => throw NotFoundException()
+    }
+  }
+
+  def getLatestRun(datasetName: String, datasetVersion: Int): Future[Run] = {
+    runMongoRepository.getLatestRun(datasetName, datasetVersion).map {
+      case Some(run) => run
+      case None      => throw NotFoundException()
+    }
+  }
+
+  def getSplineUrl(datasetName: String, datasetVersion: Int, runId: Int): Future[String] = {
+    getRun(datasetName, datasetVersion, runId).map { run =>
+      val splineRef = run.splineRef
+      String.format(splineUrlTemplate, splineRef.outputPath, splineRef.sparkApplicationId)
```

Comment:
What happens if the `getRun` returns nothing? Probably same as line 55-57

Reply:
`getRun` doesn't return `nothing`, it throws a `NotFoundException`. Exception gets handled by  `RestExceptionHandler` and returns 404. There's a bunch of integration tests covering this.

Reply:
I mixed local `getRun` with `RunMongoRepository`'s `getRun`.